src/app/highlight-2.directive.ts

```diff
@@ -0,0 +1,21 @@
+import { Directive, ElementRef, HostListener } from '@angular/core';
+
+@Directive({
+  selector: '[appHighlight2]'
+})
+export class Highlight2Directive {
+
+    @HostListener('click')
+    onClick() {
+      this.highlight();
+    }
+
+  constructor(private el: ElementRef) { }
+
+
+
+   private highlight():void {
+    this.el.nativeElement.classList.toggle('bgColor');
```

Comment:
ponieważ realizujesz to za pomocą dyrektywy to fajnie jest wykorzystać Renderer2 https://angular.io/api/core/Renderer2 który pomoże dodać operacje na stylach i w ten sposób jakby zamykasz odpowiedzialność wewnątrz dyrektywy za nadanie kolorów do podświetlenia. Możesz tutaj też wykorzystać CSS variables aby było bardziej dynamicznie. 

W obecnej formie masz dwie odpowiedzialności:

1. obsługa kiedy nadać klasę a kiedy nie
2. dyrektywa nie działa kiedy klasa nie jest zadeklarowana w innym miejscu